Packs/VersaDirector/Integrations/VersaDirector/VersaDirector.py

```diff
@@ -0,0 +1,3811 @@
+        )
+
+    # test organization name if provided
+    if client:
+        if organization_name := client.organization_params:
+            client.test_organization_name_request(organization_name)
+
+    return message
+
+
+# HEADING: """ MAIN FUNCTION """
+
+
+def main() -> None:
+    params: Dict[str, Any] = demisto.params()
+    args: Dict[str, Any] = demisto.args()
+    verify_certificate: bool = not params.get("insecure", False)
+    proxy = params.get("proxy", False)
+    use_basic_auth = params.get("use_basic_auth", False)
+    port = PORT_CREDENTIALS if use_basic_auth else PORT_AUTH
+    url = params.get("url", "")
+    url = url[:-1] + f":{port}" if url.endswith("/") else url + f":{port}"
+    context = get_integration_context()
+    if context.get("context"):
+        context = context.get("context")
+    username = params.get("credentials", {}).get("identifier", "")
+    password = params.get("credentials", {}).get("password", "")
+    client_id = params.get("client_id") or context.get("client_id")
+    client_secret = params.get("client_secret") or context.get("client_secret")
+    access_token = params.get("access_token") or context.get("access_token")
+    command = demisto.command()
+    demisto.debug(f"Command being called is {command}")
+
+    try:
+        # test_module functionality is disabled for Auth Token authentication
+        case_auth_token_auth = bool(
+            params.get("access_token", None) and not params.get("client_id", None) and not params.get("client_secret", None)
+        )
+        if command == "test-module" and not use_basic_auth and not case_auth_token_auth:
+            return_results(
+                test_module(
+                    client=None,
+                    use_basic_auth=use_basic_auth,
+                    client_id=params.get("client_id"),
+                    client_secret=params.get("client_secret"),
+                    access_token=params.get("access_token"),
+                    username=username,
+                    password=password,
+                )
+            )
+
+        auth, headers = create_client_header(use_basic_auth, username, password, client_id, client_secret, access_token)
+
+        client = Client(
+            server_url=url,
+            verify=verify_certificate,
+            headers=headers,
+            proxy=proxy,
+            auth=auth,
+            organization_params=params.get("organization"),
+            client_id_param=params.get("client_id"),
+            client_secret_param=params.get("access_token"),
+        )
+
+        # check auth token validity and if a refresh token is needed to obtain new auth token
+        if not use_basic_auth:
+            if new_token := check_and_update_token(client, client_id, client_secret, context):
+                client._headers = {"Authorization": f"Bearer {new_token}"}
+
+        commands = {
+            "test-module": test_module,
```

Comment:
?

Reply:
You check if the command is test-module and call it in a different place where the commands from `commands` are called, so no need for it to be here (it can be a bit confusing as others might think it is called the same way).